lib/kennel/models/screen.rb

```diff
@@ -147,6 +148,28 @@ def widget_defaults(type)
 
         basics.merge(custom)
       end
+
+      def resolve_link(widget, key, id_map)
+        tracking_id = widget.dig(*key)
+
+        return unless link_widget?(tracking_id)
+
+        *id_path, id_key = key
+
+        monitor_path = id_path.empty? ? widget : widget.dig(*id_path)
```

Comment:
@grosser, thoughts on a cleaner way of handling this?

Reply:
nah, as clean as it gets :)